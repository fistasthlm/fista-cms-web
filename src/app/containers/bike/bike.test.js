import React from 'react';
import { shallow } from 'enzyme';
import { Map } from 'immutable';
import Bike from './bike';

jest.mock('react-router-dom', () => ({
    withRouter: params => params,
    Redirect: params => params,
}));

const setup = propOverrides => {
    const props = Object.assign({
        authenticated: true,
        bike: Map(),
        clearBike: jest.fn(),
        loadBike: jest.fn(),
        match: {
            params: {
                id: 'hej'
            }
        }
    }, propOverrides);

    return shallow(<Bike {...props} />);
};


describe('Render', () => {
    it('should render loader if empty bike', () => {
        const wrapper = setup();
        window.location.pathname = '/bike/123';

        expect(wrapper).toMatchSnapshot();
    });

    it('should render bike info if bike', () => {
        const wrapper = setup({
            bike: Map({
                title: 'dolan',
            })
        });
        window.location.pathname = '/bike/123';

        expect(wrapper).toMatchSnapshot();
    });

    it('should render redirect if unauthenticated', () => {
        const wrapper = setup({
            authenticated: false,
        });

        window.location.pathname = '/bike/123';

        expect(wrapper).toMatchSnapshot();
    });
});

describe('Life cycle', () => {
    describe('component will unmount', () => {
        it('should clear bike', () => {
            const wrapper = setup();

            wrapper.instance().componentWillUnmount();

            expect(wrapper.instance().props.clearBike).toHaveBeenCalled();
        });
    });
});
