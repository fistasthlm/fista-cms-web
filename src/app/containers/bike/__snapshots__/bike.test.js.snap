// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render bike info if bike 1`] = `
<div
  className="edit-bike"
>
  <h1>
    Edit 
    dolan
  </h1>
  <EditBikeForm
    bike={
      Immutable.Map {
        "title": "dolan",
      }
    }
  />
</div>
`;

exports[`Render should render loader if empty bike 1`] = `<Loader />`;

exports[`Render should render redirect if unauthenticated 1`] = `
<Redirect
  to={
    Object {
      "pathname": "/login",
      "state": Object {
        "from": undefined,
      },
    }
  }
/>
`;
